### PR TITLE
Dont send small contracted invoices to finance

### PR DESCRIPTION
**Pull request by proteusvacuum**
finance@dimagi doesn't want to receive small invoices for contracted
clients any more.
http://manage.dimagi.com/default.asp?169520#972816

@nickpell 
CC @biyeun

### CODE REVIEW

File: corehq/apps/accounting/invoicing.py [hunk 1]
```diff
@@ -215,6 +215,12 @@ def create_invoice_for_subscription(self, subscription):
 
         return invoice
 
+    def should_send_email(self, invoice, subscription):
```
Comment (nickpell) [inline corehq/apps/accounting/invoicing.py]: Should this function also take ```do_not_invoice``` into account?

Comment (proteusvacuum) [inline corehq/apps/accounting/invoicing.py]: yes. will add that in a bit.

**Comment by nickpell** [comment]
test failure looks related

File: corehq/apps/accounting/tests/test_invoice_factory.py [hunk 2]
```diff
@@ -1,7 +1,16 @@
 import datetime
+from decimal import Decimal
```
Comment (dimagimon) [inline corehq/apps/accounting/tests/test_invoice_factory.py]: F401 'Decimal' imported but unused

File: corehq/apps/accounting/tests/test_invoice_factory.py [hunk 3]
```diff
@@ -1,7 +1,16 @@
...
 from corehq.apps.accounting import generator
 from corehq.apps.accounting.invoicing import DomainInvoiceFactory
-from corehq.apps.accounting.models import DefaultProductPlan, BillingAccount, Subscription, SubscriptionAdjustment
+from corehq.apps.accounting.models import (
```
Comment (dimagimon) [inline corehq/apps/accounting/tests/test_invoice_factory.py]: F401 'Invoice' imported but unused

Comment (dimagimon) [inline corehq/apps/accounting/tests/test_invoice_factory.py]: F401 'SMALL_INVOICE_THRESHOLD' imported but unused

Comment (dimagimon) [inline corehq/apps/accounting/tests/test_invoice_factory.py]: F401 'SubscriptionType' imported but unused

**Comment by proteusvacuum** [comment]
I think this is better.

File: corehq/apps/accounting/models.py [hunk 4]
```diff
@@ -1822,6 +1833,15 @@ def text_template(self):
         else:
             return self.INVOICE_TEXT_TEMPLATE
 
+    @property
+    def should_send_email(self):
+        subscription = self.invoice.subscription
+        autogenerate = (subscription.auto_generate_credits and not self.invoice.balance)
```
Comment (nickpell) [inline corehq/apps/accounting/models.py]: (nonblocking) this variable name could be clearer (it took me some time to figure out what was going on here)

**Pull request merged**